pkg/scheduler/plugins/minruntime/resolver.go

```diff
@@ -0,0 +1,241 @@
+
+			// For different top-level queues:
+			// The LCA is the implicit shadow parent at level -1 (doesn't exist in path)
+			// So we'd want to look at the top-level queue (first in path) that's the ancestor
+			// of the preemptee, which is preempteePath[0]
+
+			// Check if the top-level preemptee queue has reclaim-min-runtime set
+			if topLevelPreemptee.ReclaimMinRuntime != nil {
+				mr.cacheReclaimMinRuntime(preemptorQueue, preempteeQueue, *topLevelPreemptee.ReclaimMinRuntime)
+				return *topLevelPreemptee.ReclaimMinRuntime
+			}
+
+			// No value in hierarchy, use default
+			mr.cacheReclaimMinRuntime(preemptorQueue, preempteeQueue, *mr.defaultReclaimMinRuntime)
+			return *mr.defaultReclaimMinRuntime
+		}
+	}
+
+	// Find the lowest common ancestor (LCA)
+	lcaIndex := 0
+	minLength := min(len(preemptorPath), len(preempteePath))
+
+	for i := 0; i < minLength; i++ {
+		if preemptorPath[i].UID != preempteePath[i].UID {
+			break
+		}
+		lcaIndex = i
+	}
+
+	// If we found a common ancestor
+	if lcaIndex < len(preemptorPath) && lcaIndex < len(preempteePath) &&
+		preemptorPath[lcaIndex].UID == preempteePath[lcaIndex].UID {
+
+		lca := preemptorPath[lcaIndex]
+
+		// If there's a path down from LCA to preemptee
+		if lcaIndex+1 < len(preempteePath) {
+			// Walk 1 step down to the child of the LCA that is an ancestor to the preemptee's leaf queue
+			childQueue := preempteePath[lcaIndex+1]
+
+			// Check if this queue has reclaim-min-runtime set
+			if childQueue.ReclaimMinRuntime != nil {
+				mr.cacheReclaimMinRuntime(preemptorQueue, preempteeQueue, *childQueue.ReclaimMinRuntime)
+				return *childQueue.ReclaimMinRuntime
+			}
+
+			// Otherwise move back up towards root and select first available override
+			for i := lcaIndex; i >= 0; i-- {
+				if preempteePath[i].ReclaimMinRuntime != nil {
+					mr.cacheReclaimMinRuntime(preemptorQueue, preempteeQueue, *preempteePath[i].ReclaimMinRuntime)
+					return *preempteePath[i].ReclaimMinRuntime
+				}
+			}
+		} else {
+			// If LCA is the preemptee queue itself, check its reclaim-min-runtime
+			if lca.ReclaimMinRuntime != nil {
+				mr.cacheReclaimMinRuntime(preemptorQueue, preempteeQueue, *lca.ReclaimMinRuntime)
+				return *lca.ReclaimMinRuntime
+			}
+
+			// Otherwise move up towards root
+			for i := lcaIndex - 1; i >= 0; i-- {
+				if preempteePath[i].ReclaimMinRuntime != nil {
+					mr.cacheReclaimMinRuntime(preemptorQueue, preempteeQueue, *preempteePath[i].ReclaimMinRuntime)
+					return *preempteePath[i].ReclaimMinRuntime
+				}
+			}
+		}
+	}
+
+	// If no reclaim-min-runtime is found in the hierarchy, use default
+	mr.cacheReclaimMinRuntime(preemptorQueue, preempteeQueue, *mr.defaultReclaimMinRuntime)
+	return *mr.defaultReclaimMinRuntime
+}
+
+// getQueueHierarchyPath returns the queue hierarchy from top-level queue to the specified queue
+// Note: For top-level queues (empty parent), the path will only contain the queue itself
+func (mr *minruntimePlugin) getQueueHierarchyPath(
```

Comment:
Some thoughts - not necessarily in the scope of this PR:
Finding a common ancestor for queues will probably be useful in the core scheduler. We can save the hierarchy path for queues on snapshot instead of having the plugin do it. We can also implement the logic of finding the common ancestor in the core scheduler